pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTAnnotation.java

```diff
@@ -9,45 +9,47 @@
 import org.checkerframework.checker.nullness.qual.Nullable;
 
 import net.sourceforge.pmd.lang.ast.NodeStream;
-import net.sourceforge.pmd.util.StringUtil;
+import net.sourceforge.pmd.lang.java.symbols.JClassSymbol;
 
 /**
  * Represents an annotation.
  *
  * <pre class="grammar">
  *
- * Annotation ::= "@" Name {@link ASTAnnotationMemberList AnnotationMemberList}?
+ * Annotation ::= "@" {@link ASTClassOrInterfaceType ClassName} {@link ASTAnnotationMemberList AnnotationMemberList}?
  *
  * </pre>
  */
 public final class ASTAnnotation extends AbstractJavaTypeNode implements TypeNode, ASTMemberValue, Iterable<ASTMemberValuePair> {
 
-    String name;
-
     ASTAnnotation(int id) {
         super(id);
     }
 
 
     /**
-     * Returns the name of the annotation as it is used,
-     * eg {@code java.lang.Override} or {@code Override}.
+     * Returns the node that represents the name of the annotation.
      */
-    public String getAnnotationName() {
```

Comment:
Would it make sense to keep a deprecated `getAnnotationName()` method around (DeprecatedUntil700)?